assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -193,10 +179,43 @@ const sendFormSubmitEventForPayPalRecurring = () =>
     onFormSubmit(formSubmitParameters);
   };
 
-function setValueAndTogglePayPal<T>(setStateValue: T => Action, value: T) {
-  return (dispatch: Function): void => {
+
+function enableOrDisableForm(formIsValid: boolean, state: State, dispatch: Function) {
+
+  const shouldEnable = checkoutFormShouldSubmit({
+    formValid: formIsValid,
+    canContributeWithoutSigningIn: canContributeWithoutSigningIn(
+      state.page.form.contributionType,
+      state.page.user.isSignedIn,
+      state.page.form.userTypeFromIdentityResponse,
+    ),
+    isRecurringContributor: state.page.user.isRecurringContributor,
+  });
+
+  if (shouldEnable && window.enablePayPalButton) {
+    dispatch(setFormIsSubmittable(true));
+    window.enablePayPalButton();
```

Comment:
could we move these inside the actions? so that whenever the state is changed the PayPal button will be in sync